reference-implementation/lib/writable-stream.js

```diff
@@ -654,7 +644,17 @@ function WritableStreamDefaultWriterRelease(writer) {
 function WritableStreamDefaultWriterWrite(writer, chunk) {
   const stream = writer._ownerWritableStream;
 
-  assert(stream !== undefined);
+  if (stream === undefined) {
+    return Promise.reject(defaultWriterLockException('write to'));
+  }
+
+  const controller = stream._writableStreamController;
+
+  const chunkSize = WritableStreamDefaultControllerGetChunkSize(controller, chunk);
+
```

Comment:
Drive-by comment: maybe we should check that `stream === writer._ownerWritableStream` before proceeding?

strategySize might have
1. Detached the writer
2. Created a new writer and done a bunch of writes with it

It's probably better just to eliminate the possibility rather than try to work out whether it would be a problem or not.

Needs test(s).